presto-spi/src/main/java/io/prestosql/spi/predicate/TupleDomain.java

```diff
@@ -434,12 +434,16 @@ else if (isNone()) {
 
     public <U> TupleDomain<U> transform(Function<T, U> function)
     {
-        if (domains.isEmpty()) {
-            return TupleDomain.none();
+        if (isNone()) {
+            return none();
+        }
+        if (isAll()) {
+            return all();
         }
 
-        HashMap<U, Domain> result = new LinkedHashMap<>(domains.get().size());
-        for (Map.Entry<T, Domain> entry : domains.get().entrySet()) {
+        Map<T, Domain> domains = this.domains.orElseThrow();
```

Comment:
just use `.get()`? It cannot be empty at this point

Reply:
IntelliJ doesn't know that because of the change to that remove the `if (domains.isEmpty())` block.